fortls/parsers/internal/ast.py

```diff
@@ -36,6 +36,12 @@ def __init__(self, file_obj=None):
         self.inherit_objs: list = []
         self.linkable_objs: list = []
         self.external_objs: list = []
+        self.fold_patterns = []
+        self.folding_start: list = []
+        self.folding_end: list = []
+        self.lines_to_fold: list = []
+        self.comment_block_start = 0
+        self.comment_block_end = 0
```

Comment:
I haven't had a chance to look at this in detail but just make sure you don't duplicate information.
A lot of start/end line/column info is already stored, so you can infer them on-the-fly.